lua/entities/gmod_wire_expression2/core/custom/collcallback.lua

```diff
@@ -0,0 +1,190 @@
+E2Lib.RegisterExtension("collcallback", true)
+
+local maxcollclk = CreateConVar( "wire_col_detector_e2_maxcollclk", 100, 0, "Maximum amount of props with collision detection per gate" )
+local collallow = CreateConVar( "wire_col_detector_e2_collallow", 1, 0, "Allow collision detection of other players' stuff" )
+
+local registered_ents = {}
+
+registerCallback("construct",function(self)
+	self.colDatas = {}
+	self.colEnts = {}
+	self.collFilter = true
+	self.collIgnoreConstrained = true
+	self.colProps=0
+end)
+
+hook.Add("EntityRemoved", "Raf2EntityRemoved",function (ent)
+	registered_ents[ent]=nil
+end)
+
+function entitiesCollide(ent,data)
+	for i = 1,#registered_ents[ent],1 do
+		self = registered_ents[ent][i]
+		if not IsValid(self.entity) then
+			table.remove(registered_ents[ent],i)
+			i = i-1
+		else
+			if (self.collFilter and #self.colDatas ~= 0) or (self.collIgnoreConstrained and table.HasValue(constraint.GetAllConstrainedEntities(ent),data.HitEntity)) then return end
+			self.colData=data
+			self.colData.ourEnt = ent
+			self.colData.posFirst = ent:GetPhysicsObject():GetPos()
+			self.colData.posSecond = data.HitEntity:GetPhysicsObject():GetPos()
+			self.colData.angleFirst = ent:GetPhysicsObject():GetAngles()
+			self.colData.angleSecond = data.HitEntity:GetPhysicsObject():GetAngles()
+			if self.physicsEarlyCallback then
+				self.entity:Execute()
+			else
+				table.insert(self.colDatas,data)
+				timer.Simple(0,function()
+					self.colData = self.colDatas[1]
+					self.entity:Execute()
+					table.remove(self.colDatas,1)
+					self.colData = nil
+				end)
+			end
+			self.colData = nil
```

Comment:
Its okay because:

1. There is a function to prevent executing e2 multiple times in a tick by a single prop (enabled by default)
2. All registered collisions are saved in a table until they are used by an e2